pkg/servicemesh/controller/controller.go

```diff
@@ -104,18 +105,9 @@ type serviceMeshMemberRollListener struct {
 }
 
 func (smmrl *serviceMeshMemberRollListener) checkEquality(lhs, rhs []string) bool {
-	if (lhs == nil) != (rhs == nil) {
-		return false
-	}
-	if len(lhs) != len(rhs) {
-		return false
-	}
-	for n, val := range lhs {
-		if val != rhs[n] {
-			return false
-		}
-	}
-	return true
+	lset := sets.NewString(lhs...)
+	rset := sets.NewString(rhs...)
+	return lset.Equal(rset)
```

Comment:
It would be better to handle this in updateNamespaces, then you only need to sort the new list

Reply:
okay, I now applied the bigger change from MAISTRA-1012 here as well; we're storing a `sets.String` instead of `[]string` now, for quicker comparison. I'm sorting the array before logging, otherwise the log output is confusing - we log once per component and they would then come out ordered differently:
```
2019-10-22T14:23:56.612773Z	info	ServiceMeshMemberRoll default added, namespaces now ["test7" "test20" "test2" "test23" "test9" "test10" "test19" "test22" "test18" "test6" "test11" "test21" "test4" "test24" "test" "cp" "test8" "test5"]
2019-10-22T14:23:56.612813Z	info	ServiceMeshMemberRoll default added, namespaces now ["test2" "test23" "test7" "test19" "test20" "test4" "test9" "test18" "cp" "test11" "test22" "test10" "test" "test5" "test6" "test21" "test24" "test8"]
2019-10-22T14:23:56.612837Z	info	ServiceMeshMemberRoll default added, namespaces now ["test10" "test24" "test5" "test7" "test8" "test19" "test11" "test21" "test23" "test9" "test6" "cp" "test20" "test2" "test4" "test22" "test18" "test"]
```

Reply:
This seems to be much larger than is necessary, the only required change would be to sort the incoming array before doing the comparison.  I think you can do this with one or two lines.

Reply:
Ah. Sorry, was a misunderstanding. That was basically the first iteration of this PR. I can do it that way

Reply:
No worries, I think we can do this with just a call to sort.Strings within the updateNamespaces method.  We should only need to do this with the new list as the older one would already be sorted.

Reply:
That's how I did it now. Quickly confirmed it's all good on my repro cluster as well